Code/Framework/AzCore/Platform/Common/Apple/AzCore/Debug/Trace_Apple.cpp

```diff
@@ -27,7 +27,7 @@ namespace AZ
             // running under the debugger or has a debugger attached post facto).
             bool IsDebuggerPresent()
             {
-                int                 junk;
+                [[maybe_unused]] int                 junk;
```

Comment:
This is being used in the `assert(junk == 0);` on line 52 so it complains in profile builds.

Reply:
The variable name `junk` seems a little insulting.
How about `sysctlResult`?

Also this code was written in a very C89 style manner where variables had to be declared at the top of the function.

We can just initialize this variable where it is assigned.

Reply:
Done.